File: FrontEnd/src/app/features/billing/bankAccounts/classes/dtos/bankAccount-read-dto.ts
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface BankAccountReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    bank: SimpleEntity
    shipOwner: SimpleEntity
    // Fields
    iban: string
    isActive: boolean

}

File: API/Features/Billing/BankAccounts/Dtos/BankAccountReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Billing.BankAccounts {

    public class BankAccountReadDto : IMetadata {

        // PK
        public int Id { get; set; }
        // FKs
        public int BankId { get; set; }
        public int ShipOwnerId { get; set; }
        // Fields
        public string Iban { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }
        // Navigation
        public SimpleEntity Bank { get; set; }
        public SimpleEntity ShipOwner { get; set; }

    }

}